core/PruManager.cpp

```diff
@@ -0,0 +1,153 @@
+/*
+ * PruManager.cpp
+ * This code currently only works on BBAI
+ *
+ *	Created on: July 3, 2021
+ *		Author: Dhruva Gole
+ */
+
+#include "../include/PruManager.h"
+#include "MiscUtilities.h"
+#include <iostream>
+
+PruManager::PruManager()
+{
+	// Constructor might be of use later?
+}
+
+#if ENABLE_PRU_RPROC == 1
+PruManagerRprocMmap::PruManagerRprocMmap(unsigned int pruNum, unsigned int v)
+{
+	/* constructor for initializing the necessary path variables
+	 * based on the value of pru_num to choose:
+	 * 0 for PRU1 core 0
+	 * 1 for PRU1 core 1
+	 * 2 for PRU2 core 0
+	 * 3 for PRU2 core 1
+	 */
+	pru_num = pruNum;
+	verbose = v;
+	unsigned int pruss = pru_num / 2 + 1;
+	unsigned int prucore = pru_num % 2;
+
+	basePath = "/dev/remoteproc/pruss" + std::to_string(pruss) + "-core" + std::to_string(prucore) + "/";
+	statePath = basePath + "state";
+	firmwarePath = basePath + "firmware";
+	firmware = "am57xx-pru" + std::to_string(pruss) + "_" + std::to_string(prucore) + "-fw";
+	firmwareCopyCommand = "sudo rm /lib/firmware/" + firmware + ";sudo ln -s /home/debian/Bela-dhruva/pru/blinkR30.pru1_0.out /lib/firmware/" + firmware; // **NOTE**: Change the name of .out file according afterwards here and below cases.
```

Comment:
build this string programmatically, do not hardcode the `/home/debian/Bela-dhruva/...` path. Maybe set a define in the `Makefile` for what this path should be. I think you can use `ln -s -f` instead  of `rm` + `ln`.

Reply:
>build this string programmatically
umm I definitely could, but is that really necessary? There is always only going to be that `pru_rt_audio_irq.p` file right? This is currently done only at a testing stage. Maybe next couple of days I will see if running some example the rproc part is atleast able to flash the PRU correctly or not.